mitiq/shadows/shadows_utils.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+import cirq
+from typing import Tuple,List, Union
+#given error of the fidelity between the true state and the reconstructed state, return the number of measurements needed
+def min_n_total_measurements(epsilon: float, num_qubits: int) -> int:
+    """
+    Calculate the number of measurements required to satisfy the shadow bound for the Pauli measurement scheme.
+    
+    Args:
+        epsilon (float): The error on the estimator.
+        num_qubits (int): The number of qubits in the system.
+    
+    Returns:
+        An integer that gives the number of samples required to satisfy the shadow bound.
+    """
+    return int(34* (4**num_qubits) * epsilon**(-2))
+
+
+# based on the theorem, we calculate N,K for the shadow bound
+def shadow_bound_theorem(error: float, observables: list, failure_rate: float)->Tuple[int,int]:
```

Comment:
I would rename this function to match more closely the output (something like "calculate shadow bound")

Reply:
thanks, have renamed